### ISSUE TITLE
Refactor computation of means for Game linear/logistic regression models

### ISSUE THREAD

**Issue by ferd36** [linkedin/photon-ml#208]
Problem context and statement:

In Game, the "submodels" do not carry their own loss functions: they are trained one by one, but with a global loss function that includes data about the states of all the other submodels. 

Currently, the submodels LinearRegressionModel and LogisticRegressionModel have "computeMean" methods (including e.g. link function for logistic regression), which do not make sense in Game (the loss function is global). 

This needs to be simplified. 

Design: TBD.
